internal/testutils/stats/test_metrics_recorder.go

```diff
@@ -104,6 +104,16 @@ func (r *TestMetricsRecorder) PollForDataForMetric(ctx context.Context, metricNa
 	r.t.Fatalf("Timeout waiting for data %v for metric %v", wantVal, metricName)
 }
 
+// ClearMetrics clears the metrics data stores of the test metrics recorder by
+// setting all the data to 0.
```

Comment:
That is what I do in my constructor, but it seems like it doesn't matter because the writes in RecordSomething() will write a new data type. Should I get rid of that logic in constructor too and here?

Reply:
Oh I see.. Yes, I would just create an empty map unless there's a need to do otherwise (and then...question whether that need is truly necessary).  The zero value as the default should make everything "just work".

Reply:
Yeah fair. Switched to just making an empty map, and letting new map write take care of it. Made no sense to configure with metrics name list I guess.